R/monthly.R

```diff
@@ -1,234 +1,234 @@
-#' kijang_emas()
-#' kijang_emas(date = "2018-01-03")
-#' kijang_emas(year = 2016, month = 2)
-#' @source https://api.bnm.gov.my/
-#'
-#'
-kijang_emas <- function(date = NULL,
-                        year = NULL,
-                        month = NULL) {
-  # if date does not work most likely there's
-  # just no records for that date. write tryCatch here.
-  .get_monthly_data(
-    stub = "/kijang-emas",
-    date = date,
-    year = year,
-    month = month
-  )
-}
-
-#' USD Interbank Intraday Rate
-#'
-#' This function allows you to obtain USD/MYR
-#' interbank intraday highest and lowest rate from the BNM API.
-#' Rates are obtained from the best U.S. dollar
-#' against Malaysian ringgit interbank highest
-#' and lowest dealt rates by commercial banks on the specific date.
-#' @inheritParams islamic_interbank_rate
-#' @keywords api
-#' @export
-#' @examples
-#' usd_interbank_intraday_rate()
-#' usd_interbank_intraday_rate(date = "2018-01-03")
-#' usd_interbank_intraday_rate(year = 2016, month = 2)
-#' @source https://api.bnm.gov.my/
-#'
-#'
-usd_interbank_intraday_rate <- function(date = NULL,
-                                        year = NULL,
-                                        month = NULL) {
-  .get_monthly_data(
-    stub = "/usd-interbank-intraday-rate",
-    date = date,
-    year = year,
-    month = month
-  )
-}
-
-
-#' KL USD Reference Rate
-#'
-#' This function allows you to obtain a reference rate
-#' that is computed based on weighted average volume
-#' of the interbank USD/MYR FX spot rate transacted by
-#' the domestic financial institutions and published
-#' daily at 3:30 p.m from the BNM API.
-#' @inheritParams islamic_interbank_rate
-#' @keywords api
-#' @export
-#' @examples
-#' kl_usd_reference_rate()
-#' kl_usd_reference_rate(date = "2018-01-03")
-#' kl_usd_reference_rate(year = 2016, month = 2)
-#' @source https://api.bnm.gov.my/
-#'
-#'
-
-kl_usd_reference_rate <- function(date = NULL,
-                                  year = NULL,
-                                  month = NULL) {
-  .get_monthly_data(
-    stub = "/kl-usd-reference-rate",
-    date = date,
-    year = year,
-    month = month
-  )
-}
+#' @importFrom purrr map_dfr
+#' @importFrom jsonlite flatten
+#' @importFrom rlang list2 qq_show
+
+.get_monthly_data <- function(stub,
+                              date = NULL,
+                              year = NULL,
+                              month = NULL, ...) {
+
+  # function should consider length 1 for arguments?
+  args <- list(...)
+
+
+  if (is.null(date)) {
+    if (is.null(month)) {
+      if (is.null(year)) {
+        get_bnm_data(glue("{stub}"), query = args)
+      } else {
+        map_dfr(1:12, function(x) {
+          Sys.sleep(1)
+          flatten(get_bnm_tbl(glue("{stub}/year/{year}/month/{x}"),
+                              query = args))
+            }
+          )
+      }
+
+
+    } else if (!is.null(year)) {
+      stopifnot(is.numeric(year) & is.numeric(month))

```

Comment:
```suggestion
      stopifnot(is.numeric(year) && is.numeric(month))

```

Reply:
we should expect year and month of length 1 here, I was trying to extend the functionality to accept vectors but from line 30 I didn't quite get there for this function